### DISCUSSION TITLE
How to check relative simplicity of a polygon?

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#7249] [Q&A] [answered]
In the context of *2D regularized Boolean operations*, the outer polygon of a polygon with holes must be relatively simple in order to be considered as valid. I didn't find a `is_relatively_simple` function with the search input. How to check this property? That does not seem trivial.

**Comment by efifogel** [comment, marked as answer]
[ola] CGAL > git grep is_relatively_simple_polygon
      1 Boolean_set_operations_2/Gps_polygon_validation.h:bool
is_relatively_simple_polygon
      2 Boolean_set_operations_2/Gps_polygon_validation.h:bool
is_relatively_simple_polygon_with_holes
      3 Boolean_set_operations_2/Gps_polygon_validation.h:  bool
outer_is_relatively_simple =
is_relatively_simple_polygon(pgn.outer_boundary(), traits);
      4 Boolean_set_operations_2/Gps_polygon_validation.h:    bool
relatively_simple_hole = is_relatively_simple_polygon(*hoit, traits);
      5 Boolean_set_operations_2/Gps_polygon_validation.h:  bool
relatively_simple = is_relatively_simple_polygon_with_holes(pgn, traits);
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by stla** [comment]
:+1:  thanks!

**Comment by stla** [comment]
There's even `CGAL::is_valid_polygon_with_holes`, better :+1:
